controllers/dashboards/dashboard_controller.go

```diff
@@ -35,9 +36,23 @@ import (
 var dlog = logger.WithField("module", "dashboard_controller")
 
 func (r *PersesDashboardReconciler) reconcileDashboardInAllInstances(ctx context.Context, req ctrl.Request) (*ctrl.Result, error) {
+	dashboard := &persesv1alpha1.PersesDashboard{}
+
+	if r, err := r.getLatestPersesDashboard(ctx, req, dashboard); subreconciler.ShouldHaltOrRequeue(r, err) {
+		return r, err
+	}
+
+	labelSelector, err := metav1.LabelSelectorAsSelector(dashboard.Spec.InstanceSelector)
+	if err != nil {
+		return subreconciler.RequeueWithError(err)
+	}
+
 	persesInstances := &persesv1alpha1.PersesList{}
-	var opts []client.ListOption
-	err := r.Client.List(ctx, persesInstances, opts...)
+	opts := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+
+	err = r.Client.List(ctx, persesInstances, opts)
```

Comment:
will this still list Perses instances when no label selectors were configured?

Reply:
yes, it should select all instances when the label selector is empty. Also I updated one of the tests to define it a little more and kept the other test to not specifying an Instance selector at all and they both seem fine still.